src/engraving/dom/note.cpp

```diff
@@ -1405,6 +1405,70 @@ void Note::updateFrettingForTiesAndBends()
     setFret(prevNote->fret());
 }
 
+bool Note::hideFret() const
+{
+    if (!tieBack() || forceShowFret()) {
+        return false;
+    }
+
+    ShowTiedFret showTiedFret = style().value(Sid::tabShowTiedFret).value<ShowTiedFret>();
+    if (showTiedFret == ShowTiedFret::TIE_AND_FRET) {
+        return false;
+    }
+
+    if (showTiedFret == ShowTiedFret::TIE) {
+        return true;
+    }
+
+    ParenthesizeTiedFret parenthTiedFret = style().value(Sid::tabParenthesizeTiedFret).value<ParenthesizeTiedFret>();
+    if (parenthTiedFret == ParenthesizeTiedFret::NEVER || !rtick().isZero()) {
+        return true;
+    }
+
+    if (parenthTiedFret == ParenthesizeTiedFret::START_OF_MEASURE) {
+        return false;
+    }
+
+    const Measure* measure = findMeasure();
+
+    return !(measure && measure->isFirstInSystem());
+}
+
+bool Note::forceShowFret() const
+{
+    if (!style().styleB(Sid::parenthesizeTiedFretIfArticulation)) {
+        return false;
+    }
+
+    ShowTiedFret showTiedFret = style().value(Sid::tabShowTiedFret).value<ShowTiedFret>();
+    Chord* ch = chord();
+    if (showTiedFret != ShowTiedFret::NONE || !ch) {
+        return false;
+    }
+
+    auto hasTremoloBar = [&] () {
+        for (EngravingItem* item : ch->segment()->annotations()) {
+            if (item && item->isTremoloBar()) {
```

Comment:
Is it intentional that you're not checking the track or staff index here?

Reply:
Good point, thanks 👍